packages/integration-tests/src/e2e-flow/e2e_flow.test.ts

```diff
@@ -0,0 +1,258 @@
+        StackName: BackendIdentifierConversions.toStackName(
+          branchBackendIdentifier
+        ),
+      })
+    );
+    await fsp.rm(tempDir, { recursive: true });
+    // stop the npm proxy
+    await execa('npm', ['run', 'stop:npm-proxy'], { stdio: 'inherit' });
+  });
+
+  void it(`starting project`, async () => {
+    await execa(
+      packageManager.startsWith('yarn') ? 'yarn' : packageManager,
+      ['create', 'amplify', '--yes'],
+      {
+        cwd: tempDir,
+        stdio: 'inherit',
+      }
+    );
+
+    const amplifyPathPrefix = path.join(tempDir, 'amplify');
+
+    const pathPrefix = path.join(tempDir, 'amplify');
+
+    const files = await glob(path.join(amplifyPathPrefix, '**', '*'), {
+      // eslint-disable-next-line spellcheck/spell-checker
+      nodir: true,
+      windowsPathsNoEscape: true,
+      ignore: ['**/node_modules/**', '**/yarn.lock'],
+    });
+
+    const expectedAmplifyFiles = [
+      path.join('auth', 'resource.ts'),
+      'backend.ts',
+      path.join('data', 'resource.ts'),
+      'package.json',
+      'tsconfig.json',
+    ];
+
+    assert.deepStrictEqual(
+      files.sort(),
+      expectedAmplifyFiles.map((suffix) => path.join(pathPrefix, suffix))
+    );
+
+    if (packageManager === 'yarn-modern') {
+      await execa('yarn', ['config', 'set', 'nodeLinker', 'node-modules'], {
+        cwd: `${tempDir}/amplify`,
+        stdio: 'inherit',
+      });
+
+      await fsp.appendFile(
+        path.join(tempDir, '.yarnrc.yml'),
+        `pnpIgnorePatterns:\n  - ./nm-packages/**`
+      );
+      await execa('yarn', ['install'], {
+        cwd: tempDir,
+        stdin: 'inherit',
+      });
+      await execa('yarn', ['add', '@aws-amplify/backend'], {
+        cwd: `${tempDir}/amplify`,
+        stdio: 'inherit',
+      });
+    }
+
+    // assert that project compiles successfully
+    await execa(
+      packageManagerExecutable,
+      [
+        'tsc',
+        '--noEmit',
+        '--skipLibCheck',
+        // pointing the project arg to the amplify backend directory will use the tsconfig present in that directory
+        '--project',
+        amplifyPathPrefix,
+      ],
+      {
+        cwd: tempDir,
+        stdio: 'inherit',
+      }
+    );
+
+    if (packageManager.startsWith('yarn')) {
+      await execa('yarn', ['add', 'aws-cdk', 'aws-cdk-lib', 'constructs'], {
+        cwd: tempDir,
+        stdio: 'inherit',
+      });
+      if (packageManager === 'yarn-modern') {
+        await execa(
+          'yarn',
+          [
+            'add',
+            '-D',
+            'tsx',
+            'graphql',
+            'pluralize',
+            'zod',
+            '@aws-amplify/platform-core',
+            'esbuild',
+          ],
+          {
+            cwd: tempDir,
+            stdio: 'inherit',
+          }
+        );
+
+        await execa('node', ['--version'], {
+          cwd: tempDir,
+        });
+      }
+    }
+
+    if (packageManager === 'pnpm') {
```

Comment:
This can be removed since it doesn't fix `pnpm` failure on windows

Reply:
The `pnpm` failure is because it doesn't suppor nested node_modules. [ref](https://pnpm.io/faq)